File: spring-boot-kafka-demo/src/main/java/com/github/jccode/springbootkafkademo/event/EventConsumer.java
package com.github.jccode.springbootkafkademo.event;

import org.apache.kafka.clients.consumer.ConsumerRecord;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
public class EventConsumer {

    @KafkaListener(topics = Constant.EVENT_TOPIC)
    public void listenEvents(ConsumerRecord<String, Event> record) {
        Event value = record.value();
        Object payload = value.getPayload();
        System.out.println("Receive event: "+value.getType()+", "+payload);
    }
}

File: spring-boot-kafka-demo/kafka-json-demo1/src/main/scala/com/github/jccode/springbootkafkademo/jsondemo12/event/EventConsumer.scala
package com.github.jccode.springbootkafkademo.jsondemo12.event

import org.apache.kafka.clients.consumer.ConsumerRecord
import org.slf4j.{Logger, LoggerFactory}
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component

@Component
class EventConsumer {

  private val logger: Logger = LoggerFactory.getLogger(classOf[EventConsumer])

  @KafkaListener(topics = Array(Constant.EventTopic))
  def listenEvents(record: ConsumerRecord[String, Event]): Unit = {
    val value = record.value()
    logger.info(s"Receive event: ${value.eventType}, ${value.payload}")
  }
}
